### PR TITLE
Adds stdio Support to rosparam

### PR DESCRIPTION
**Pull request by evenator**
`rosparam dump` to stdout is useful for piping the rosparam yaml output to other programs. `rosparam load` from stdin is useful for piping yaml into rosparam.

This PR implements both, such that if the user does not specify a filename to the command (or specifies "-" as the filename), `rosparam` uses stdio instead of creating a file handle.

My particular usecase for `rosparam load` from stdin is being able to only load parameters from a launch file:

```
roslaunch robot.launch --dump-params | rosparam load -
```

### CODE REVIEW

File: tools/rosparam/src/rosparam/__init__.py [hunk 1]
```diff
@@ -160,11 +160,16 @@ def load_file(filename, default_namespace=None, verbose=False):
       corresponding namespaces for each YAML document in the file
     :raises: :exc:`RosParamException`: if unable to load contents of filename
     """
-    if not os.path.isfile(filename):
-        raise RosParamException("file [%s] does not exist"%filename)
-    if verbose:
-        print("reading parameters from [%s]"%filename)
-    f = open(filename, 'r')
+    if filename == '-' or not len(filename):
```
Comment (dirk-thomas) [inline tools/rosparam/src/rosparam/__init__.py]: You can use `not filename` directly without the need to call `len()`.

Same below.

File: tools/rosparam/src/rosparam/__init__.py [hunk 2]
```diff
@@ -495,12 +503,12 @@ def _rosparam_cmd_set_load(cmd, argv):
     if cmd == 'set':
         if options.text_file and options.bin_file:
             parser.error("you may only specify one of --textfile or --binfile")
-
     arg2 = None
+
```
Comment (dirk-thomas) [inline tools/rosparam/src/rosparam/__init__.py:506]: Please avoid unrelated white space changes.

File: tools/rosparam/src/rosparam/__init__.py [hunk 3]
```diff
@@ -495,12 +503,12 @@ def _rosparam_cmd_set_load(cmd, argv):
...
     if len(args) == 0:
         if cmd == 'load':
-            parser.error("invalid arguments. Please specify a file name")
-        elif cmd == 'set':
+            arg = '-'
+        if cmd == 'set':
```
Comment (dirk-thomas) [inline tools/rosparam/src/rosparam/__init__.py]: Please avoid unrelated changes and keep `elif` here.

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by evenator** [comment]
Apparently the CI server chokes on rebased branches. Should I close this PR and open a new one with the recommended changes?

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by evenator** [comment]
Ok, looks like it's passing now.

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder

**Comment by evenator** [comment]
I'm just going to try this again...
